astropy/coordinates/representation/base.py

```diff
@@ -377,14 +382,34 @@ def _apply(self, method, *args, **kwargs):
         return new
 
     def __setitem__(self, item, value):
-        if value.__class__ is not self.__class__:
+        set_mask = value is np.ma.masked
+        clear_mask = value is np.ma.nomask
+        if not (value.__class__ is self.__class__ or set_mask or clear_mask):
             raise TypeError(
```

Comment:
I think the error message needs to be updated.

Reply:
Will do (after waiting a bit more to check if there are further comments)

Reply:
I rebased to try to get reviews before feature freeze and fixed this one in the process.